File: gdbm/python/read/gdbm_read.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	gdbm_read.py
#
#					Jun/11/2015
import	sys
import json
#import anydbm
import dbm
import string
#
sys.path.append ('/var/www/data_base/common/python_common')
from dbm_manipulate import dbm_to_dict_proc
from text_manipulate import dict_display_proc
# -------------------------------------------------------------
print ("*** 開始 ***")
db_name = sys.argv[1]
# dd = anydbm.open (db_name,"c")
dd = dbm.open (db_name,"c")
#
dict_aa = dbm_to_dict_proc (dd)
#
dd.close ()
#
dict_display_proc (dict_aa)
#
print ("*** 終了 ***")
# -------------------------------------------------------------

File: gdbm/ruby/read/gdbm_read.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	read/gdbm_read.rb
#
#				Aug/26/2013
#
# --------------------------------------------------------------------
require	'dbm'
require	'json'
#
load '/var/www/data_base/common/ruby_common/kvalue_manipulate.rb'
# --------------------------------------------------------------------
puts	"*** 開始 ***"
#
file_gdbm=ARGV[0]
ee=DBM.open(file_gdbm)
ww=ee.to_hash;
ee.close
#
kvalue_display_proc (ww)
#
print "ww.size =",ww.size,"\n"
#
puts "*** 終了 ***"
#
# --------------------------------------------------------------------
